src/plugins/tts/FestivalTTSEngine.js

```diff
@@ -122,21 +121,18 @@ class FestivalTTSSound {
         if (this.rate != 1) this.sound.setPlaybackRate(this.rate);
         onload();
       },
-      onresume: () => {
-        sleep(25).then(() => {
-          if (this.rate != 1) this.sound.setPlaybackRate(this.rate);
-        });
+      onresume: async () => {
+        await sleep(25);
+        if (this.rate != 1) this.sound.setPlaybackRate(this.rate);
       }
     });
     return this.sound.load();
   }
 
-  play() {
-    return new Promise(res => {
-      this._finishResolver = res;
-      this.sound.play({ onfinish: res });
-    })
-      .then(() => this.sound.destruct());
+  async play() {
+    const destruct = await this.sound.destruct();
+    this._finishResolver = destruct;
+    this.sound.play({ onfinish: destruct});
```

Comment:
Similarly here; we'll steel need the `new Promise` to promisify the `onfinish` event.